addons/web/static/src/notifications/notification.xml

```diff
@@ -20,7 +20,7 @@
             <div t-if="props.message" t-raw="props.message" class="mr-auto o_notification_content"/>
             <div t-if="props.buttons.length" class="mt-2 o_notification_buttons">
                 <button t-foreach="props.buttons" t-as="button" type="button"
-                        t-attf-class="btn btn-sm {{button.primary ? 'btn-primary' : 'btn-secondary'}}">
+                        t-attf-class="btn btn-sm {{button.primary ? 'btn-primary' : 'btn-secondary'}}" t-on-click="button.onClick()">
```

Comment:
I think you don't have to call it, `t-on-click="button.onClick"` should be sufficient